pkg/machines/components/vm/vmActions.jsx

```diff
@@ -17,110 +17,164 @@
+                {_("Send Non-Maskable Interrupt")}
+            </DropdownItem>
+        );
+        dropdownItems.push(<DropdownSeparator key="separator-sendnmi" />);
     }
 
-    let resume = null;
-    if (LibvirtDBus.canResume(state)) {
-        resume = (<button key='action-resume' className="pf-c-button pf-m-secondary" onClick={mouseClick(onResume)} id={`${id}-resume`}>
-            {_("Resume")}
-        </button>);
+    if (LibvirtDBus.canReset(state)) {
+        dropdownItems.push(
+            <DropdownItem key={`${id}-reboot`}
+                          id={`${id}-reboot`}
+                          onClick={() => onReboot()}>
+                {_("Restart")}
+            </DropdownItem>
+        );
+        dropdownItems.push(
+            <DropdownItem key={`${id}-forceReboot`}
+                          id={`${id}-forceReboot`}
+                          onClick={() => onForceReboot()}>
+                {_("Force Restart")}
+            </DropdownItem>
+        );
+        dropdownItems.push(<DropdownSeparator key="separator-reset" />);
     }
 
     let run = null;
     if (LibvirtDBus.canRun(state, hasInstallPhase)) {
-        run = (<button key='action-run' className="pf-c-button pf-m-secondary" onClick={mouseClick(onStart)} id={`${id}-run`}>
+        run = (<Button key='action-run' variant="secondary" onClick={() => onStart()} id={`${id}-run`}>
             {_("Run")}
-        </button>);
+        </Button>);
     }
 
     let install = null;
     if (LibvirtDBus.canInstall(state, hasInstallPhase)) {
-        install = (<button key='action-install' className="pf-c-button pf-m-secondary" onClick={mouseClick(onInstall)} id={`${id}-install`}>
+        install = (<Button key='action-install' variant="secondary" onClick={() => onInstall()} id={`${id}-install`}>
             {_("Install")}
-        </button>);
+        </Button>);
     }
 
     let deleteAction = null;
     if (state !== undefined && LibvirtDBus.canDelete && LibvirtDBus.canDelete(state, vm.id)) {
-        deleteAction = (
-            <DeleteDialog key='action-delete' vm={vm} dispatch={dispatch} storagePools={storagePools} />
+        dropdownItems.push(
+            <DropdownItem className='pf-m-danger' key={`${id}-delete`} id={`${id}-delete`} onClick={() => toggleDeleteModal(true)}>
+                {_("Delete")}
+            </DropdownItem>
         );
+
+        if (!vm.persistent) {
+            dropdownItems.push(
+                <DropdownItem key={`${id}-delete`}
+                              id={`${id}-delete`}
+                              variant='danger'
+                              tooltip={<Tooltip id={`${id}-delete-tooltip`}>
+                                  {_("This VM is transient. Shut it down if you wish to delete it.")}
+                              </Tooltip>}
+                              isDisabled>
+                    {_("Delete")}
+                </DropdownItem>
+            );
+        }
```

Comment:
This needs to be restyled. Like this, first delete button is pushed into array and rendered always, and if vm is persistend, then second Delete link is also rendered. 
Checkout the [screenshot](https://logs.cockpit-project.org/logs/pull-13474-20200426-185359-a3c0d528-centos-8-stream/TestMachinesDBus-testDelete-centos-8-stream-127.0.0.2-2201-FAIL.png) of this situation.
I think correct solution should be something like:
```
        if (vm.persistent) {
            dropdownItems.push(
￼                 <DropdownItem className='pf-m-danger' key={`${id}-delete`} id={`${id}-delete`} onClick={() => toggleDeleteModal(true)}>
￼                     {_("Delete")}
￼                 </DropdownItem>
￼             );
￼         } else {
￼             dropdownItems.push(
￼                 <DropdownItem key={`${id}-delete`}
￼                               id={`${id}-delete`}
￼                               variant='danger'
￼                               tooltip={<Tooltip id={`${id}-delete-tooltip`}>
￼                                   {_("This VM is transient. Shut it down if you wish to delete it.")}
￼                               </Tooltip>}
￼                               isDisabled>
￼                     {_("Delete")}
￼                 </DropdownItem>
￼             );
￼         }
```

Reply:
We also need to adjust the order.

The non-maskable interrupt should be right before delete. I think the order we want would be:

> - Pause
> --
> - Shutdown
> - Force shut down
> --
> - Restart
> - Force restart
> --
> - Misc. stuff here
> --
> - Delete

So the order should be:

1. Common actions
2. Miscellaneous, rarely used
3. _Danger, [Will Robinson](https://cultural-phenomenons.fandom.com/wiki/Danger,_Will_Robinson)!_